### PR TITLE
feat(policy): add MeshTCPRoute api with validation

### PR DESCRIPTION
**Pull request by bartsmykla**
Add `MeshTCPRoute` api with validation

### Checklist prior to review

- [x] [Link to relevant issue][1] as well as docs and UI issues 
  - https://github.com/kumahq/kuma/issues/6787
- [x] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS
  - it won't
- [x] Tests (Unit test, E2E tests, manual test on universal and k8s)
- [x] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)?
  - it's a new policy, so there is no need
- [x] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? 
  - it's a new policy, so there is no need
- [x] Do you need to explicitly set a [`> Changelog:` entry here](../blob/master/CONTRIBUTING.md#submitting-a-patch) or add a `ci/` label to run fewer/more tests?
  - nope

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: pkg/plugins/policies/meshtcproute/api/v1alpha1/validator.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+func (r *MeshTCPRouteResource) validate() error {
+	var verr validators.ValidationError
+
+	path := validators.RootedAt("spec")
+
+	verr.AddErrorAt(path.Field("targetRef"), validateTop(r.Spec.TargetRef))
+
+	if len(r.Spec.To) == 0 {
+		verr.AddViolationAt(path.Field("to"), "needs at least one item")
```
Comment (michaelbeaumont) [inline pkg/plugins/policies/meshtcproute/api/v1alpha1/validator.go:17]: Could just be `MinItems`?

File: pkg/plugins/policies/meshtcproute/api/v1alpha1/validator.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+		verr.AddErrorAt(path.Field("rules"), validateRules(toItem.Rules))
+	}
+
+	return verr
+}
+
+func validateRules(rules []Rule) validators.ValidationError {
+	var verr validators.ValidationError
+
+	if len(rules) != 1 {
```
Comment (michaelbeaumont) [inline pkg/plugins/policies/meshtcproute/api/v1alpha1/validator.go:55]: This is already validated by the kubebuilder annotations

Comment (bartsmykla) [reply pkg/plugins/policies/meshtcproute/api/v1alpha1/validator.go]: Good point: addressed

File: pkg/plugins/policies/meshtcproute/api/v1alpha1/validator.go (hunk 1 above)
Comment (bartsmykla) [reply pkg/plugins/policies/meshtcproute/api/v1alpha1/validator.go]: Good point: addressed

**Review by michaelbeaumont — APPROVED** [review]
LGTM

**Pull request merged**